### PR TITLE
upgrade to babel 7

### PR DESCRIPTION
**Pull request by tzimmermann**
Trying to solve #36

What I have done:

- ~~Moved `babel-core` (_only_ used to bundle the signavio-18n package) from dep. to devDep.~~
  - `@babel/core` is now _both_, devDep (for bundling) and dep (for runtime string extraction). Both is using only v7 now.
- Make the getText extractor script compatible with babel 7.

Problems:

- ~~The peerDep definition in the `package.json` for babel 6 is not really working. Before the 7 release, the package name was not namespaced (but `babel-core`).~~
  - Dropped babel 6 support, so this is solved.
- Testing this locally is a pain because, when doing a local installation `yarn add path/to/signavio-i18n`, yarn will simply copy the folder, which includes the `babel-core` devDep., which breaks stuff.
- I could only test this locally by 
  - running `yarn && yarn build`
  - removing `node_modules`
  - doing `yarn --prod` (to skip installing devDeps)
  - `yarn add path/to/signavio-i18n` in my project
  - `yarn i18n` in my project (tested with babel v6 & v7 installed).

### CODE REVIEW

**Comment by jfschwarz** [comment]
Ok, just drop support for babel-core@6 then. So we will have `@babel/core` as a peer and dev dep

File: .babelrc [hunk 1]
```diff
@@ -1,34 +1,61 @@
 {
   "plugins": [
-    "transform-proto-to-assign",
-    "transform-flow-strip-types",
+    "@babel/plugin-transform-proto-to-assign",
```
[the remark this answers is not in the document]
Comment (tzimmermann) [reply .babelrc]: Good point, fixed that. I didn't even realise both configs were nearly identical 😆

File: .babelrc.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    '@babel/plugin-transform-runtime',
+    'transform-remove-console',
+    'transform-remove-debugger',
+    'transform-minify-booleans',
+    '@babel/plugin-syntax-dynamic-import',
+    '@babel/plugin-syntax-import-meta',
+    '@babel/plugin-proposal-class-properties',
+    '@babel/plugin-proposal-json-strings',
+    [
+      '@babel/plugin-proposal-decorators',
```
Comment (jfschwarz) [inline .babelrc.js]: Why are we adding all these babel plugins? I don't think we are using decorators syntax in this repository, are we?

Comment (jfschwarz) [reply .babelrc.js]: I'd think we should be able to significantly simplify the transpilation setup for this package. Just use some babel presets

Comment (tzimmermann) [reply .babelrc.js]: Well, the whole idea of babel 7 is to remove the `stage-x` or the yearly presets in favor of explicitly using the individual proposals. See:

https://babeljs.io/docs/en/v7-migration#yearly-preset-deprecations-blog-2017-12-27-nearing-the-70-releasehtml-deprecated-yearly-presets-eg-babel-preset-es20xx

So far, the 6->7 migration automatically added all plugins that were included in "es2015", "stage-2", "react" that we used before.  
But yes, we surely can remove some of them that are not in use.

**Comment by jfschwarz** [comment]
Ah, this is a result of an automated migration! In that case, it‘s cool. We can strip down the configuration in a subsequent PR then.

**Comment by tzimmermann** [comment]
@jfschwarz @frontendphil Anything left to do, or can we merge & release this?

**Comment by frontendphil** [comment]
@tzimmermann released as `2.0.0`

**Pull request merged**